ENG-6593: Introduce support for stream event cursors (#168)
fauna/fauna-go: README.md

`Stream()`. To override these options, see [query
options](#query-options).

The `Subscribe()` method accepts the `fauna.StartTime` and `fauna.EventCursor`
function. Use `fauna.StartTime` to restart a stream at a specific timestamp.

```go
streamQuery, _ := fauna.FQL(`Product.all().toStream()`, nil)
client.Subscribe(streamQuery, fauna.StartTime(1710968002310000))
```

Use `fauna.EventCursor` to resume a stream after a disconnect:

```go
streamQuery, _ := fauna.FQL(`Product.all().toStream()`, nil)
client.Subscribe(streamQuery, fauna.EventCursor("abc2345=="))
```

| Function | Description |
| -------- | ----------- |
| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |

> | `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` with the cursor retrieved from a `fauna.Event`. |

Reviewer: ```suggestion
| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` representation of the cursor retrieved from a `fauna.Event`. |
```

`Stream()`. To override these options, see [query
options](#query-options).

The `Subscribe()` method accepts the `fauna.StartTime` and `fauna.EventCursor`

> function. You can use `fauna.StartTime` to restart a stream at a specific
> timestamp.

Reviewer 2: ```suggestion
function. Use `fauna.StartTime` to restart a stream at a specific
timestamp:
```

```go
streamQuery, _ := fauna.FQL(`Product.all().toStream()`, nil)
client.Subscribe(streamQuery, fauna.StartTime(1710968002310000))
```

The `fauna.EventCursor` function can be used to resume a stream after a

> disconnect.

Reviewer 2: ```suggestion
Use `fauna.EventCursor` to resume a stream after a disconnect:

```go
streamQuery, _ := fauna.FQL(`Product.all().toStream()`, nil)
client.Subscribe(streamQuery, fauna.EventCursor("abc2345=="))
```

| Function | Description |
| -------- | ----------- |
| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |

> | `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts an `string` with the cursor retrieved from a `fauna.Event`. |

Reviewer 2: ```suggestion
| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` with the cursor retrieved from a `fauna.Event`. |
```
